### ISSUE TITLE
question about early stopping

### ISSUE THREAD

**Issue by DopamineLcy** [QingruZhang/AdaLoRA#3]
Thank you for your marvelous work!
Did you use a validation set and early stopping to select the best-performing checkpoint? If so, how do you ensure that the budget reaches the target budget?
Best,

**Comment by QingruZhang** [comment]
Hi, thanks for your question. We did not apply early stopping. We evaluate the model during the final fine-tuning stages, when AdaLoRA reaches the target budget and fix it. The checkpoint can be selected during the final fine-tuning stage, i.e. after `total_step-final_warmup` steps. Hope this can answer your questions.

**Comment by A11en0** [comment]
Why not use an early stop, otherwise it may not find the best checkpoint. And Adalora seems not select the best checkpoint in the final fine-tuning stage.
